airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py

```diff
@@ -230,6 +248,10 @@ def _send_request(self, request: requests.PreparedRequest, request_kwargs: Mappi
         Unexpected persistent exceptions are not handled and will cause the sync to fail.
         """
         response: requests.Response = self._session.send(request, **request_kwargs)
+        if not self.raise_on_http_errors:
```

Comment:
not clear how and what circumstance we should use it.

1. I would ask you to prepare a few "use cases", present and discuss them on the nearest KT. 
2. and discuss how it implement comments:
- Let's go with option 3 and find a way to handle backoff reasonably  (Option 3: don't fail on any HTTP error, let the user decide inside parse_response)
- In certain cases like the Asana API, an API might return a 4XX that can be remedied/fixed. For example, the API returns a 412 if the input query parameter sync is invalid, and the response will contain a new sync token that should be used. In this case, the request should not be considered a failure, and instead should be retried with the new parameter.

Reply:
you should use `raise_on_http_errors` if you need to process responses with http error codes.
in asana case you can receive `412` response, get new token from it and update `next_page_token` or `stream_state`